File: evaluations/gridcal/verify_install.py
"""Verify GridCal installation by running DC power flow on IEEE 39-bus case."""

from __future__ import annotations

import sys
from pathlib import Path

import GridCalEngine as gce

DATA_DIR = Path(__file__).resolve().parent.parent.parent / "data" / "networks"

# Load case39.m
grid = gce.open_file(str(DATA_DIR / "case39.m"))

# Run DC power flow
results = gce.power_flow(grid, engine=gce.EngineType.DC)

print(f"GridCalEngine version: {gce.__version__}")
print(f"Buses: {grid.get_bus_number()}")
print(f"Branches: {grid.get_branch_number()}")
print(f"Converged: {results.converged}")
sys.exit(0 if results.converged else 1)

File: evaluations/powermodels/verify_install.jl
"""
Verify PowerModels.jl installation by running DC power flow on IEEE 39-bus case.
"""

using PowerModels
using HiGHS

data_dir = joinpath(@__DIR__, "..", "..", "data", "networks")
case_file = joinpath(data_dir, "case39.m")

# Parse MATPOWER case file
data = PowerModels.parse_file(case_file)
println("Network: $(data["name"])")
println("Buses: $(length(data["bus"]))")
println("Branches: $(length(data["branch"]))")

# Run DC power flow
result = PowerModels.solve_dc_pf(data, HiGHS.Optimizer)
println("Status: $(result["termination_status"])")
println("Solve time: $(result["solve_time"])s")

if result["termination_status"] == OPTIMAL
    println("DC power flow completed successfully")
    exit(0)
else
    println("DC power flow failed")
    exit(1)
end
